realm-annotations/src/main/java/io/realm/annotations/NotNullable.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2014 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.annotations;
+
+public @interface NotNullable {
```

Comment:
How so? Eg. IntelliJ's @NotNull is a hint that this String can never be null, exactly the same as ours. I agree that if you want to use multiple definitions you have to be a bit careful, which could lead to very subtle errors:

```
public class Foo {
  @NotNull
  private String bar;

  private void setBar(@com.intellij.annotations.NotNulll bar) {
    this.bar = bar;
  }
}
```

Here are some other options: http://stackoverflow.com/questions/4963300/which-notnull-java-annotation-should-i-use

Another radical solution: We still have default values and renaming as upcoming features. Would it makes sense instead to do something like

````
@RealmField(defaultValue = "foo", nullable = false, name="internal_name_used_in_tables_and_queries") 
```